### DISCUSSION TITLE
How to run a public site

### DISCUSSION THREAD

**Discussion by IvanIsak2000** [Spaceblock/cosmoterm#6] [General]
When we launch the site on a local PC - this is understandable, but we need to decide where we will launch the public site, I suggest `ngrok` - the application makes the local site public with a random domain (if on a free plan)

https://ngrok.com

**Comment by mb6ockatf** [comment]
> When we launch the site on a local PC - this is understandable, but we need to decide where we will launch the public site, I suggest `ngrok` - the application makes the local site public with a random domain (if on a free plan)
> 
> https://ngrok.com

sounds great)
as an option, i can host it at home on my raspberry PI. Although, hosting is not a problem really - yet, as we haven't enough content on out website

**Reply by IvanIsak2000** [reply]
I also have a free PC on ubuntu, it can also be used as a server for the first time

**Reply by mb6ockatf** [reply]
> I also have a free PC on ubuntu, it can also be used as a server for the first time

nice :)

**Reply by mb6ockatf** [reply]
we do not need a public website, if we're developing a messager core. as we have a clear application api, everyone can develop a decent web/native interface for it

**Reply by IvanIsak2000** [reply]
Hi! It seems to me that I have developed a simple way to log in to your account:

![image](https://user-images.githubusercontent.com/79650307/235614306-284220cd-5827-4de4-9506-ad0ae4fb6027.png)
![image](https://user-images.githubusercontent.com/79650307/235614387-0bd380cd-2127-440b-8c3f-1b47de1ddbdb.png)

![image](https://user-images.githubusercontent.com/79650307/235614484-7a1b36e1-322b-48f8-bdc3-c9991d71394d.png)

Im proud of myself !

**Reply by IvanIsak2000** [reply]
> Hi! It seems to me that I have developed a simple way to log in to your account:
> 
> ![image](https://user-images.githubusercontent.com/79650307/235614306-284220cd-5827-4de4-9506-ad0ae4fb6027.png) ![image](https://user-images.githubusercontent.com/79650307/235614387-0bd380cd-2127-440b-8c3f-1b47de1ddbdb.png)
> 
> ![image](https://user-images.githubusercontent.com/79650307/235614484-7a1b36e1-322b-48f8-bdc3-c9991d71394d.png)
> 
> Im proud of myself !

The fact is that the page <random domain from ngrok>/log_in/username+usertoken is pre-registered

**Reply by IvanIsak2000** [reply]
> > Hi! It seems to me that I have developed a simple way to log in to your account:
> > ![image](https://user-images.githubusercontent.com/79650307/235614306-284220cd-5827-4de4-9506-ad0ae4fb6027.png) ![image](https://user-images.githubusercontent.com/79650307/235614387-0bd380cd-2127-440b-8c3f-1b47de1ddbdb.png)
> > ![image](https://user-images.githubusercontent.com/79650307/235614484-7a1b36e1-322b-48f8-bdc3-c9991d71394d.png)
> > Im proud of myself !
> 
> The fact is that the page /log_in/username+usertoken is pre-registered

If there is only one user - one html page with status 200, but if there are more of them, you will have to make a database
I think I need a list from db, which will be substituted in the request url
![image](https://user-images.githubusercontent.com/79650307/235615299-94f55ec9-b5bc-40a6-805a-108ab58044e4.png)

**Reply by mb6ockatf** [reply]
> > > Hi! It seems to me that I have developed a simple way to log in to your account:
> > > ![image](https://user-images.githubusercontent.com/79650307/235614306-284220cd-5827-4de4-9506-ad0ae4fb6027.png) ![image](https://user-images.githubusercontent.com/79650307/235614387-0bd380cd-2127-440b-8c3f-1b47de1ddbdb.png)
> > > ![image](https://user-images.githubusercontent.com/79650307/235614484-7a1b36e1-322b-48f8-bdc3-c9991d71394d.png)
> > > Im proud of myself !
> > 
> > The fact is that the page /log_in/username+usertoken is pre-registered
> 
> If there is only one user - one html page with status 200, but if there are more of them, you will have to make a database
> I think I need a list from db, which will be substituted in the request url
> ![image](https://user-images.githubusercontent.com/79650307/235615299-94f55ec9-b5bc-40a6-805a-108ab58044e4.png)
> 

that's a bad idea. you should process such data in request params, not in url

**Reply by IvanIsak2000** [reply]
> > > > Hi! It seems to me that I have developed a simple way to log in to your account:
> > > > ![image](https://user-images.githubusercontent.com/79650307/235614306-284220cd-5827-4de4-9506-ad0ae4fb6027.png) ![image](https://user-images.githubusercontent.com/79650307/235614387-0bd380cd-2127-440b-8c3f-1b47de1ddbdb.png)
> > > > ![image](https://user-images.githubusercontent.com/79650307/235614484-7a1b36e1-322b-48f8-bdc3-c9991d71394d.png)
> > > > Im proud of myself !
> > > 
> > > 
> > > The fact is that the page /log_in/username+usertoken is pre-registered
> > 
> > 
> > If there is only one user - one html page with status 200, but if there are more of them, you will have to make a database
> > I think I need a list from db, which will be substituted in the request url
> > ![image](https://user-images.githubusercontent.com/79650307/235615299-94f55ec9-b5bc-40a6-805a-108ab58044e4.png)
> 
> that's a bad idea. you should process such data in request params, not in url

It was just an experiment

**Reply by mb6ockatf** [reply]
oh, i'm sorry

**Comment by IvanIsak2000** [comment]
This is the updated code

```
#server.py
from fastapi import FastAPI, Body, status, Request
from fastapi.responses import HTMLResponse


app = FastAPI()

fake_data = {'Iwan':'123'}

@app.get("/log_in/{username}/{usertoken}/")
def log_in(username: str, usertoken: str):
        
        if fake_data[username] == usertoken:
                return True
        else:
                return False

```


```
#client.py
import sys
import os 
import requests
from tqdm import tqdm
import time 

def log_in(username: str, usertoken: str)-> bool:

    succesfull_connect = False

    URL = RANDOM  URL FROM NGROK' #this random domen i get from ngrok

    response = requests.get(f'{URL}/log_in/{username}/{usertoken}')
    status = response.status_code

    if response.content ==  b'true':
        succesfull_connect = True
        

    if succesfull_connect :
        os.system('cls')
        print('Welcome!')

    else:
        print('not valid data!')
        print(f'Статус запроса {status}')

try:

    name = str(sys.argv[1])
    token = str(sys.argv[2])
    
    log_in(name, token)

except  Exception as err:
    print(f"{err}\nВы должна ввести правильно: <username>  <usertoken>")




````
